type(Kubernetes): Error 400 using TCP proxy and nginx ingress (#3448)
scaleway/docs-content: containers/kubernetes/reference-content/kubernetes-load-balancer.mdx

To force the Load Balancer to handle all requests using the SSL offload, enable the [use hostname annotations](https://github.com/scaleway/scaleway-cloud-controller-manager/blob/master/docs/loadbalancer-annotations.md#servicebetakubernetesioscw-loadbalancer-use-hostname) on your Load Balancer service.

### Load Balancer TCP Proxy and Nginx Ingress error 400

If your Load Balancer is configured using TCP Proxy and you are using an Nginx Ingress, you might encounter an HTTP 400 error.

The Nginx Ingress must be configured with [use-proxy-protocol](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-proxy-protocol) to accept TCP Proxy requests.

#### Nginx Ingress Easy Deploy

The [Easy Deploy](/containers/kubernetes/how-to/enable-easy-deploy) Nginx Ingress can be configured to accept TCP Proxy:
```

> controller:
>   config:
>     use-proxy-protocol: 'true'

Reviewer: ```suggestion
controller:
  service:
    annotations:
      service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"
      service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"
  config:
    use-forwarded-headers: true
    compute-full-forwarded-for: true
    use-proxy-protocol: true
```

To force the Load Balancer to handle all requests using the SSL offload, enable the [use hostname annotations](https://github.com/scaleway/scaleway-cloud-controller-manager/blob/master/docs/loadbalancer-annotations.md#servicebetakubernetesioscw-loadbalancer-use-hostname) on your Load Balancer service.

### Load Balancer TCP Proxy and Nginx Ingress error 400

> If your Load Balancer is configured using TCP Proxy and and you are using an Nginx Ingress, you might encounter an http 400 error.

Reviewer 2: ```suggestion
If your Load Balancer is configured using a TCP Proxy and you are using an Nginx Ingress, you might encounter an HTTP 400 error.
```

The Nginx Ingress must be configured with [use-proxy-protocol](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-proxy-protocol) to accept TCP Proxy requests.

> #### Nginx Ingress EasyDeploy

Reviewer 3: ```suggestion
#### Nginx Ingress Easy Deploy
```

> This configuration is the default one when using the [EasyDeploy](/containers/kubernetes/how-to/enable-easy-deploy) Nginx Ingress.

Reviewer 3: ```suggestion
This configuration is the default one when using the [Easy Deploy](/containers/kubernetes/how-to/enable-easy-deploy) Nginx Ingress.
```

> This configuration is the default one when using the [Easy Deploy](/containers/kubernetes/how-to/enable-easy-deploy) Nginx Ingress.

Reviewer 4: Proxy protocol on the loadbalancer is not enabled by default on nginx ingress from easydeploy. adding config bellow alone (without corresponding annotation) would break the ingress-lb communication.

Author: I checked and you are right, by default there is no proxy.

What is better ? give the "complete" Easy Deploy configuration for having a LB TCP proxy that works with nginx (annotation + config), or delete this subSection ?

> The EasyDeploy Nginx can be configured to accept TCP Proxy:

Reviewer 3: ```suggestion
The Easy Deploy Nginx can be configured to accept TCP Proxy:
```
